File: frontend/src/types/availability.ts
export interface Availability {
  id: number;
  day_of_week: number;
  start_time: string;
  end_time: string;
  created_at: string;
  updated_at: string;
}

export interface AvailabilityCreateInput {
  day_of_week: number;
  start_time: string;
  end_time: string;
}

export type AvailabilityUpdateInput = AvailabilityCreateInput;

export const WEEKDAYS = [
  { value: 0, label: 'Monday' },
  { value: 1, label: 'Tuesday' },
  { value: 2, label: 'Wednesday' },
  { value: 3, label: 'Thursday' },
  { value: 4, label: 'Friday' },
  { value: 5, label: 'Saturday' },
  { value: 6, label: 'Sunday' },
] as const;

File: backend/app/schemas/availability.py
from datetime import time, datetime

from pydantic import BaseModel, Field, ConfigDict, model_validator

class AvailabilityBase(BaseModel):
    """
    Base schema for availability blocks.
    """

    day_of_week: int = Field(
        ...,
        ge=0,
        le=6,
        description="0 = Monday ... 6 = Sunday",
    )

    start_time: time
    end_time: time

    @model_validator(mode="after")
    def validate_time_range(self):
        if self.start_time >= self.end_time:
            raise ValueError("start_time must be before end_time")

        return self


class AvailabilityCreate(AvailabilityBase):
    pass


class AvailabilityUpdate(AvailabilityBase):
    pass


class AvailabilityResponse(BaseModel):
    """
    Response schema for availability blocks.
    """

    id: int
    day_of_week: int
    start_time: time
    end_time: time
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )
